oracle/price-feeder/oracle/oracle_test.go

```diff
@@ -350,6 +353,218 @@ func (ots *OracleTestSuite) TestPrices() {
+				{Base: "BTC", ChainDenom: "ubtc", Quote: "USD"},
+				{Base: "OTHER", ChainDenom: "uother", Quote: "USD"}, // filtered out
+			},
+			prices: map[string]sdk.Dec{
+				"USDT":  sdk.MustNewDecFromStr("1.1"),
+				"BTC":   sdk.MustNewDecFromStr("2.2"),
+				"OTHER": sdk.MustNewDecFromStr("3.3"),
+			},
+			whitelist: denomList("uusdt", "ubtc"),
+			expectedVoteMsg: &oracletypes.MsgAggregateExchangeRateVote{
+				ExchangeRates: "2.200000000000000000ubtc,1.100000000000000000uusdt", // does not include uother
+				Feeder:        feederAddr,
+				Validator:     validatorAddr,
+			},
+		},
+		"Jailed should return error": {
+			isJailed:    true,
+			blockHeight: 1,
+			expectedErr: fmt.Errorf("validator %s is jailed", validatorAddr),
+		},
+		"Zero block height should return error": {
+			isJailed:    false,
+			blockHeight: 0,
+			expectedErr: fmt.Errorf("expected positive block height"),
+		},
+	}
+
+	ctx := context.Background()
+	for testName, tc := range tests {
+		test := tc
+		cdm, _ := createMappingsFromPairs(test.pairs)
+		t.Run(testName, func(t *testing.T) {
+			var setPriceCount int
+			var broadcastCount int
+			// Create the oracle instance
+			oracle := &Oracle{
+				jailCache: JailCache{
+					isJailed: test.isJailed,
+				},
+				mockSetPrices: func(ctx context.Context) error {
+					setPriceCount++
+					return nil
+				},
+				chainDenomMapping: cdm,
+				prices:            test.prices,
+				paramCache: ParamCache{
+					params: &oracletypes.Params{
+						Whitelist: test.whitelist,
+					},
+				},
+				oracleClient: client.OracleClient{
+					OracleAddrString:    feederAddr,
+					ValidatorAddrString: validatorAddr,
+					MockBroadcastTx: func(ctx sdkclient.Context, msgs ...sdk.Msg) (*sdk.TxResponse, error) {
+						// Assert that there's only one message
+						require.Equal(t, 1, len(msgs))
+
+						// Extract the message of type MsgAggregateExchangeRateVote
+						voteMsg, ok := msgs[0].(*oracletypes.MsgAggregateExchangeRateVote)
+						require.True(t, ok, "Expected message type *oracletypes.MsgAggregateExchangeRateVote")
+
+						// Assert the expected values in the voteMsg
+						require.Equal(t, test.expectedVoteMsg.ExchangeRates, voteMsg.ExchangeRates)
+						require.Equal(t, test.expectedVoteMsg.Feeder, voteMsg.Feeder)
+						require.Equal(t, test.expectedVoteMsg.Validator, voteMsg.Validator)
+
+						broadcastCount++
+						return &sdk.TxResponse{TxHash: "0xhash", Code: 200}, nil
+					},
+				},
+			}
+
+			// execute the tick function
+			err := oracle.tick(ctx, sdkclient.Context{}, test.blockHeight)
+
+			if test.expectedErr != nil {
+				require.Equal(t, test.expectedErr, err)
+			} else {
+				require.NoError(t, err)
+			}
+			if test.expectedVoteMsg != nil {
+				// ensure functions were actually called
+				require.Equal(t, 1, broadcastCount)
+				require.Equal(t, 1, setPriceCount)
+			}
+		})
+	}
+}
+
+func TestFilterPricesWithDenomList(t *testing.T) {
+	tests := []struct {
```

Comment:
see above comment about map[string]struct{}, i slightly prefer this array structure better